lib/inputs/go/index.ts

```diff
@@ -0,0 +1,84 @@
+import * as elf from 'elfy';
+import * as path from 'path';
+import { Readable } from 'stream';
+import { ExtractAction } from '../../extractor/types';
+
+const ignoredPaths = [
+  path.normalize('/boot'),
+  path.normalize('/dev'),
+  path.normalize('/etc'),
+  path.normalize('/home'),
+  path.normalize('/media'),
+  path.normalize('/mnt'),
+  path.normalize('/proc'),
+  path.normalize('/root'),
+  path.normalize('/run'),
+  path.normalize('/sbin'),
+  path.normalize('/sys'),
+  path.normalize('/tmp'),
+  path.normalize('/var'),
+];
+
+function filePathMatches(filePath: string): boolean {
+  const dirName = path.dirname(filePath);
+  return (
+    !path.parse(filePath).ext &&
+    !ignoredPaths.some((ignorePath) => dirName.startsWith(ignorePath))
+  );
+}
+
+export const getGoModulesContentAction: ExtractAction = {
+  actionName: 'gomodules',
+  filePathMatches,
+  callback: findGoBinaries,
+};
+
+async function findGoBinaries(stream: Readable): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const result: Buffer[] = [];
+    const buildInfoMagic = '\xff Go buildinf:';
+
+    stream.on('end', () => {
+      // Discard
+      if (result.length === 0) {
+        return resolve();
+      }
+
+      const buffer = Buffer.concat(result);
+      const binaryFile = elf.parse(buffer);
+      const goBuildInfo = binaryFile.body.sections.find(section => section.name === '.go.buildinfo');
+
+      if (!goBuildInfo) {
+        return resolve();
+      } else {
+        const info = buffer.slice(goBuildInfo.off, goBuildInfo.off + 14).toString('binary');
+
+        if (info === buildInfoMagic) {
+          return resolve(binaryFile);
+        }
+
+        return resolve();
+      }
+    });
+
+    stream.on('error', (error) => reject(error));
+
+    stream.once('data', (chunk) => {
+      const elfHeaderMagic = '\x7FELF';
+      const first16Bytes = chunk.toString('binary', 0, 4);
+
+      if (first16Bytes !== elfHeaderMagic) {
+        stream.destroy();
+      }
+    });
+
+    stream.on('data', (chunk) => {
+      // It actually exists from Node.js v8.0.0
+      if ((stream as any).destroyed) {
```

Comment:
Awesome! Do we need to support earlier versions of Node? I suppose 8 is quite ancient history

Reply:
Because it's gonna be used in CLI, we need support from Node v8 and up. When we drop Node 8 support in CLI, we might update plugins.